mongo/integration/clam_prose_test.go

```diff
@@ -0,0 +1,225 @@
+			collectionName:    "aff43dfcaa1a4014b58aaa9606f5bd44",
+			maxDocumentLength: 5,
+			operation: func(ctx context.Context, mt *mtest.T, coll *mongo.Collection) {
+				result := coll.Database().RunCommand(ctx, bson.D{{"notARealCommand", true}})
+				assert.NotNil(mt, result.Err(), "expected RunCommand error, got: %v", result.Err())
+			},
+			orderedLogValidators: []logTruncCaseValidator{
+				nil,
+				newLogTruncCaseValidator(mt, "failure", func(cmd string) error {
+					if len(cmd) != 5+len(logger.TruncationSuffix) {
+						return fmt.Errorf("expected reply to be %d bytes, got %d",
+							5+len(logger.TruncationSuffix), len(cmd))
+					}
+
+					return nil
+				}),
+			},
+		},
+		//{
+		//	name:              "3 Truncation with multi-byte codepoints",
+		//	collectionName:    "41fe9a6918044733875617b56a3125a9",
+		//	maxDocumentLength: 454, // One byte away from the end of the UTF-8 sequence 世.
+		//	operation: func(ctx context.Context, mt *mtest.T, coll *mongo.Collection) {
+		//		_, err := coll.InsertOne(ctx, bson.D{{"x", "hello 世"}})
+		//		assert.Nil(mt, err, "InsertOne error: %v", err)
+		//	},
+		//	orderedLogValidators: []logTruncCaseValidator{
+		//		nil,
+		//		newLogTruncCaseValidator(mt, "reply", func(cmd string) error {
+		//			fmt.Println("cmd: ", cmd)
+		//			// Ensure that the tail of the command string is "hello ".
+		//			if !strings.HasSuffix(cmd, "hello "+logger.TruncationSuffix) {
+		//				return fmt.Errorf("expected command to end with 'hello ', got %q", cmd)
+		//			}
+
+		//			return nil
+		//		}),
+		//	},
+		//},
+	} {
+		tcase := tcase
+
+		mt.Run(tcase.name, func(mt *mtest.T) {
+			mt.Parallel()
+
+			const deadline = 1 * time.Second
+			ctx := context.Background()
+
+			sinkCtx, sinkCancel := context.WithDeadline(ctx, time.Now().Add(deadline))
+			defer sinkCancel()
+
+			validator := func(order int, level int, msg string, keysAndValues ...interface{}) error {
+				// If the order exceeds the length of the "orderedCaseValidators," then throw an error.
+				if order >= len(tcase.orderedLogValidators) {
+					return fmt.Errorf("not enough expected cases to validate")
+				}
+
+				caseValidator := tcase.orderedLogValidators[order]
+				if caseValidator == nil {
+					return nil
+				}
+
+				return tcase.orderedLogValidators[order](keysAndValues...)
+			}
+
+			sink := newTestLogSink(sinkCtx, mt, len(tcase.orderedLogValidators), validator)
+
+			// Configure logging with a minimum severity level of "debug" for the "command" component
+			// without explicitly configure the max document length.
+			loggerOpts := options.Logger().SetSink(sink).
+				SetComponentLevel(options.LogComponentCommand, options.LogLevelDebug)
+
+			if mdl := tcase.maxDocumentLength; mdl != 0 {
+				loggerOpts.SetMaxDocumentLength(mdl)
+			}
+
+			clientOpts := options.Client().SetLoggerOptions(loggerOpts).ApplyURI(mtest.ClusterURI())
+
+			client, err := mongo.Connect(context.TODO(), clientOpts)
```

Comment:
```suggestion
			client, err := mongo.Connect(context.Background(), clientOpts)
```
`context.TODO()` is only for examples.